plugins/chat/app/services/chat/lookup_channel_threads.rb

```diff
@@ -54,28 +54,87 @@ def can_view_channel(guardian:, channel:, **)
     end
 
     def fetch_threads(guardian:, channel:, **)
-      Chat::Thread
-        .strict_loading
-        .includes(
-          :channel,
-          last_reply: %i[user uploads],
-          original_message_user: :user_status,
-          original_message: [
-            :chat_webhook_event,
-            :chat_mentions,
-            :chat_channel,
-            user: :user_status,
-          ],
+      threads =
```

Comment:
I tried to avoid doing this stuff in multiple queries, but it just has way too much grouping and special conditions involved to do so, and the resulting query was not right or had way too much data returned.

Reply:
Ok, I guess in the future we can also lower the limit to something like 20 which will reduce the cost of this.